src/java.base/share/classes/jdk/internal/foreign/Utils.java

```diff
@@ -202,4 +205,35 @@ public static void checkAllocationSizeAndAlign(long byteSize, long byteAlignment
             throw new IllegalArgumentException("Invalid alignment constraint : " + byteAlignment);
         }
     }
+
+    private static long computePadding(long offset, long align) {
+        boolean isAligned = offset == 0 || offset % align == 0;
+        if (isAligned) {
+            return 0;
+        } else {
+            long gap = offset % align;
+            return align - gap;
+        }
+    }
+
+    public static StructLayout computeStructLayout(MemoryLayout... elements) {
```

Comment:
Not too sure about this - at least this being here. The general spirit of Linker classification is that padding bits are inserted as required by the client. Now, it seems you need this inside the fallback FFI, so that you can create the precise layout of a struct where some of the fields might be platform-dependent - in that case I'd say it would be better to move this function over there?

Reply:
Note that we don't use this on any of the user supplied layouts. This is only used to compute the layout of `ffi_type` in `FFIType`. On 64bit platforms this needs padding, on 32bit platforms it does not.

I can move it into `FFIType` as well, but it seemed like a more general piece of functionality (i.e. maybe we could change the CallGeneratorHelper test class to use this instead)

Reply:
Yep - I've seen that. I guess perhaps what I find confusing is the name - e.g. one might thing that MemoryLayout.structLayout should call this, but it doesn't. But I don't have any great suggestions. Perhaps I'd stress the alignment, so maybe "computePaddedStructLayout" would be clearer?